bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,447 @@
+            sendCommand(command);
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error commanding scene run: {}", e.getMessage());
+            throw new SomfyMyLinkException("Error commanding scene run", e);
+        }
+    }
+
+    private void sendCommand(String command) throws SomfyMyLinkException {
+        synchronized (CONNECTION_LOCK) {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection(); OutputStream out = socket.getOutputStream()) {
+                    byte[] sendBuffer = command.getBytes(StandardCharsets.US_ASCII);
+                    // send the command
+                    out.write(sendBuffer, 0, sendBuffer.length);
+                }
+
+                // give time for mylink to process
+                Thread.sleep(CONNECTION_DELAY);
+            } catch (SocketTimeoutException e) {
+                logger.warn("Timeout sending command to mylink: {} Message: {}", command, e.getMessage());
+                throw new SomfyMyLinkException("Timeout sending command to mylink", e);
+            } catch (IOException e) {
+                logger.warn("Problem sending command to mylink: {} Message: {}", command, e.getMessage());
+                throw new SomfyMyLinkException("Problem sending command to mylink", e);
+            } catch (InterruptedException e) {
+                logger.debug("Interrupted while waiting after sending command to mylink: {} Message: {}", command,
+                        e.getMessage());
+            }
+        }
+    }
+
+    private @Nullable SomfyMyLinkResponseBase sendCommandWithResponse(String command, Type responseType)
+            throws SomfyMyLinkException {
+        synchronized (CONNECTION_LOCK) {
+            try {
+                logger.debug("Sending: {}", command);
+                try (Socket socket = getConnection();
+                        OutputStream out = socket.getOutputStream();
+                        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
+                    byte[] sendBuffer = command.getBytes(StandardCharsets.US_ASCII);
+
+                    // send the command
+                    out.write(sendBuffer, 0, sendBuffer.length);
+
+                    // now read the reply
+                    char[] readBuff = new char[1024];
+                    int readCount;
+                    String message = "";
+
+                    // while we are getting data ...
+                    while (((readCount = in.read(readBuff)) != -1)) {
+                        logger.debug("Got response. Len: " + readCount);
+                        message += new String(readBuff, 0, readCount);
+
+                        try {
+                            logger.debug("Got message: " + message);
+
+                            JsonParser parser = new JsonParser();
+                            JsonObject jsonObj = parser.parse(message).getAsJsonObject();
```

Comment:
Why not just parse from the Reader directly and skip the String message middleman?

Reply:
I felt having the messages as strings first would help with logging the output first, before json parsing.